From 92e785d887a8a61a897289f2b342f754cf02551b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Thomas=20Wei=C3=9Fschuh?= <thomas.weissschuh@linutronix.de>
Date: Fri, 24 Jul 2026 15:40:30 +0200
Subject: [PATCH 12785/15333] random: vDSO: Drop custom PAGE_SIZE definitions
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

These custom definitions where added in commit 7f053812dab3 ("random:
vDSO: minimize and simplify header includes") to avoid pulling in
non-vDSO-safe headers from linux/page.h.

vdso/page.h provides these definitions for the vDSO now. Use that instead.

Signed-off-by: Thomas Weißschuh <thomas.weissschuh@linutronix.de>
Signed-off-by: Thomas Gleixner <tglx@kernel.org>
Link: https://patch.msgid.link/20260724-vdso-random-page-size-v1-1-e69283d36b2f@linutronix.de
---
 lib/vdso/getrandom.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/lib/vdso/getrandom.c b/lib/vdso/getrandom.c
index 7e29005aa208..2851afa9154f 100644
--- a/lib/vdso/getrandom.c
+++ b/lib/vdso/getrandom.c
@@ -8,6 +8,7 @@
 #include <vdso/datapage.h>
 #include <vdso/getrandom.h>
 #include <vdso/limits.h>
+#include <vdso/page.h>
 #include <vdso/unaligned.h>
 #include <asm/barrier.h>
 #include <asm/vdso/getrandom.h>
@@ -18,11 +19,6 @@
 /* Bring in default accessors */
 #include <vdso/vsyscall.h>
 
-#undef PAGE_SIZE
-#undef PAGE_MASK
-#define PAGE_SIZE (1UL << CONFIG_PAGE_SHIFT)
-#define PAGE_MASK (~(PAGE_SIZE - 1))
-
 #define MEMCPY_AND_ZERO_SRC(type, dst, src, len) do {				\
 	while (len >= sizeof(type)) {						\
 		__put_unaligned_t(type, __get_unaligned_t(type, src), dst);	\
-- 
2.55.0

